feature/tpm/tpm.go

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+// Package tpm implements support for TPM 2.0 devices.
+package tpm
+
+import (
+	"slices"
+	"sync"
+
+	"github.com/google/go-tpm/tpm2"
+	"github.com/google/go-tpm/tpm2/transport"
+	"tailscale.com/feature"
+	"tailscale.com/hostinfo"
+	"tailscale.com/tailcfg"
+)
+
+var infoOnce = sync.OnceValue(info)
+
+func init() {
+	feature.Register("tpm")
+	hostinfo.RegisterHostinfoNewHook(func(hi *tailcfg.Hostinfo) {
+		hi.TPM = infoOnce()
+	})
+}
+
+//lint:ignore U1000 used in Linux and Windows builds only
+func infoFromCapabilities(tpm transport.TPM) *tailcfg.TPMInfo {
+	info := new(tailcfg.TPMInfo)
+	for _, cap := range []struct {
+		prop  tpm2.TPMPT
+		apply func(info *tailcfg.TPMInfo, value uint32)
+	}{
+		{tpm2.TPMPTManufacturer, func(info *tailcfg.TPMInfo, value uint32) { info.Manufacturer = uint32ToString(value) }},
+		{tpm2.TPMPTVendorString1, func(info *tailcfg.TPMInfo, value uint32) { info.Vendor += uint32ToString(value) }},
+		{tpm2.TPMPTVendorString2, func(info *tailcfg.TPMInfo, value uint32) { info.Vendor += uint32ToString(value) }},
+		{tpm2.TPMPTVendorString3, func(info *tailcfg.TPMInfo, value uint32) { info.Vendor += uint32ToString(value) }},
+		{tpm2.TPMPTVendorString4, func(info *tailcfg.TPMInfo, value uint32) { info.Vendor += uint32ToString(value) }},
```

Comment:
what are you trying to do here with these uint32s? what sort of strings are you building here?

you didn't document it in tailcfg.go at all

Reply:
these are 4 ASCII characters encoded as `uint32`.
each `GetCapability` request to the TPM returns a single `uint32` value, and they have different meanings:
* `TPMPTManufacturer` is a 4-character vendor code from https://trustedcomputinggroup.org/wp-content/uploads/TCG-TPM-Vendor-ID-Registry-Family-1.2-and-2.0-Version-1.07-Revision-0.02_pub.pdf (part 4.1) (e.g. `MSFT` for Microsoft)
* `TPMPTVendorString*` are 4 4-character sections of the vendor string (e.g. `Pluton.TPM.A` on my framework laptop)

I added a doc comment on decoding in the helper function below, and doc comments in tailcfg to explain the fields.